src/apps/persistence/docker_registries.clj

```diff
@@ -13,6 +14,15 @@
   (first (select docker-registries
                  (where {:name name}))))
 
+(defn get-registry-from-image
+  [image-name]
+  (let [parts (string/split image-name #"/")]
+    (loop [n (count parts)]
+      (when-not (= n 0)
+        (if-let [reg (get-registry (string/join "/" (take n parts)))]
+          reg
+          (recur (- n 1)))))))
```

Comment:
Maybe it's safe to assume that the registry will usually not have a `/` in the name?
Then we could loop the other way, progressively adding parts until we find the registry name, preventing us hitting the db too many times in most cases?
```suggestion
    (loop [n 1]
      (when (> (count parts) n)
        (if-let [reg (get-registry (string/join "/" (take n parts)))]
          reg
          (recur (+ n 1)))))))
```

Reply:
I didn't do this because I guess I was figuring we wanted the longest match, in order to allow greater specificity with a default (e.g. one set of credentials for dockerhub generally, but a different one for a specific username's images, or such). That might be another question though, if I'm overengineering it :)

I suppose if we're matching at the start, I might be able to roll it all directly into one SQL query, something in the vein of `SELECT ... WHERE $image_name LIKE name || '%'` maybe

Reply:
I see, I didn't know that was a possibility.
In that case, this looks good to me.